HW4_Voskoboinikov/ultimate_protein_tools.py

```diff
@@ -0,0 +1,691 @@
+            raise ValueError(f'Unknown elem: {chem[char]}')
+
+    return total_mass
+
+
+def convert_aa_name(sequence: str, name_dict: dict = None, sep: str = '',
+                    use_default_register: bool = True) -> str:
+    """
+    Converts a sequence of one-letter amino acid codes to three-letter designations.
+
+    Arguments:
+    - sequence (str): String with one-letter amino acid codes.
+    - name_dict (dict[str, str], optional): A dictionary linking one-letter codes to three-letter designations.
+    If not provided, the standard AA_NAME_DICT dictionary is used.
+    - sep (str, optional): Separator between three-letter amino acid designations. There is no delimiter by default.
+    - use_default_register(bool, optional): Determines whether to preserve letter case in three-letter designations.
+    If True, the letters will be converted to upper or lower case depending on the case of the depending
+    on the case of the one-letter code. The default is False.
+
+    Return:
+    - str: A string of three-letter amino acid designations separated by the specified delimiter.
+    """
+
+    new_name = ''
+    if name_dict is None:
+        name_dict = AA_NAME_DICT
+    for i, aa in enumerate(sequence):
+        if aa in name_dict:
+            if use_default_register is False:
+                new_name += name_dict[aa]
+            elif use_default_register is True:
+                if aa.isupper():
+                    new_name += name_dict[aa].upper()
+                else:
+                    new_name += name_dict[aa].lower()
+            else:
+                if aa.isupper():
+                    new_name += name_dict[aa].lower()
+                else:
+                    new_name += name_dict[aa].upper()
+            if sep and (i + 1) < len(sequence):
+                new_name += sep
+        else:
+            raise ValueError(f'Unknown amino acid: {aa}')
+    return new_name
+
+# defined later to let all funcs be initialized before passed here
+command_dct = {
+    'find_sites': find_sites,
+    'get_protein_rnas': get_protein_rnas,
+    'get_protein_rnas_number': get_protein_rnas_number,
+    'get_frameshift_proteins': get_frameshift_proteins,
+    'is_protein_valid': is_protein_valid,
+    'get_length_of_protein': get_length_of_protein,
+    'count_aa': count_aa,
+    'get_fracture_of_aa': get_fracture_of_aa,
+    'calculate_protein_mass': calculate_protein_mass,
+    'get_atomic_mass': get_atomic_mass,
+    'convert_aa_name': convert_aa_name,
+    }
+
+
+
+def parse_input(inp: str, **kwargs) -> dict:
+    """
+    Parses input and returns dict of seqs.
+
+    Arguments:
+    - inp (str): Input path or seq or dict of seqs or list of seqs
+    - **kwargs: Additional keyword arguments to be passed to input reader (e.g. )
+
+    Return:
+    - parsed_dct (dict): dict where keys are number or name of seq and value of seq
+    """
+
+    parsed_dct = {}
+    inp_type = type(inp)  # get input type
+    if inp_type == list:
```

Comment:
Лучше для проверки типа использовать isinstance()
```suggestion
    if isinstance(inp_type, list):
```